client/src/containers/Visualisation.jsx

```diff
@@ -177,16 +177,17 @@ class Visualisation extends Component {
       return <div className="Visualisation">Loading...</div>;
     }
     const { VisualisationHeader, VisualisationEditor } = this.state.asyncComponents;
+    const clonedVisualisation = cloneDeep(this.state.visualisation);
```

Comment:
Is cloning necessary here? A quick look at where we update `this.state.visualisation` via `setState` we seem to try to do it functionally via `Object.assign` or `update`.